enterprise_access/apps/api/v1/views.py

```diff
@@ -134,6 +159,119 @@ def _validate_subsidy_request(self):
             logger.exception(error_msg)
             raise SubsidyRequestCreationError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
 
+    def _validate_subscription_plan_uuid(self, subscription_plan_uuid):
+        """
+        Raises SubsidyRequestError if subsidy_request_uuid not present or invalid
+
+        subsidy_request_uuid should be a valid uuid.
+        """
+        if not subscription_plan_uuid:
+            error_msg = (
+                'You must provide a License Subscription UUID so subsidy requests '
+                'can be completed (so a License can be assigned to the user).'
+            )
+            logger.exception(error_msg)
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        try:
+            validate_uuid(subscription_plan_uuid)
+        except ParseError as exc:
+            error_msg = f'Subscription Plan UUID provided ({subscription_plan_uuid}) is not a valid UUID'
+            logger.exception(error_msg)
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY) from exc
+
+    def _verify_subsidies_remaining(self, subscription_plan_uuid, subsidy_request_uuids):
+        """
+        Check license manager to make sure there are subscriptions remaining.
+        """
+        client = LicenseManagerApiClient()
+        subscription_overview = client.get_subscription_overview(subscription_plan_uuid)
+        status_counts = subscription_overview.get('results')
+
+        remaining_unassigned_codes = 0
+        for status_count in status_counts:
+            if status_count['status'] == 'unassigned':
+                remaining_unassigned_codes = status_count['count']
+                break
+
+        if remaining_unassigned_codes < len(subsidy_request_uuids):
+            error_msg = (
+                f'Not enough licenses available for subscription {subscription_plan_uuid} '
+                'to approve the requests'
+            )
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+    @permission_required(
+        constants.REQUESTS_ADMIN_LEARNER_ACCESS_PERMISSION,
```

Comment:
I think this should be REQUESTS_ADMIN_ACCESS_PERMISSION since we only want to allow admins to approve.

Reply:
gooood call